crates/router/src/configs/settings.rs

```diff
@@ -797,7 +810,16 @@ impl Settings<SecuredSecret> {
 
         let config = router_env::Config::builder(&environment.to_string())
             .change_context(ApplicationError::ConfigurationError)?
-            .add_source(File::from(config_path).required(false))
+            .add_source(File::from(config_path).required(false));
+
+        #[cfg(feature = "v2")]
+        let config = {
+            let required_fields_path_buf =
```

Comment:
Nit: Naming, optional change. Can we call this `required_fields_config_file` instead?